checkstyle.xml

```diff
@@ -0,0 +1,122 @@
+
+    <!-- Allow individual lines of code to be excluded from these rules, if they are annotated
+         with @SuppressWarnings. See also SuppressWarningsHolder below -->
+    <module name="SuppressWarningsFilter" />
+
+    <!-- Check individual Java source files for specific rules -->
+    <module name="TreeWalker">
+        <!-- Maximum line length is 120 characters -->
+        <module name="LineLength">
+            <property name="max" value="120"/>
+        </module>
+
+        <!-- Check for and report back any TODO or FIXME comments as info messages -->
+        <module name="TodoComment">
+            <property name="severity" value="info"/>
+            <property name="format" value="(TODO)|(FIXME)"/>
+        </module>
+
+        <!-- Do not report errors on any lines annotated with @SuppressWarnings -->
+        <module name="SuppressWarningsHolder"/>
+
+        <!-- ##### Import statement requirements ##### -->
+        <!-- Star imports (e.g. import java.util.*) are NOT ALLOWED -->
+        <module name="AvoidStarImport"/>
+        <!-- Redundant import statements are NOT ALLOWED -->
+        <module name="RedundantImport"/>
+        <!-- Unused import statements are NOT ALLOWED -->
+        <module name="UnusedImports"/>
+
+        <!-- ##### Javadocs requirements ##### -->
+        <!-- Requirements for Javadocs for classes/interfaces -->
+        <module name="JavadocType">
+            <!-- All public classes/interfaces MUST HAVE Javadocs -->
+            <property name="scope" value="public"/>
+            <!-- Add an exception for anonymous inner classes -->
+            <property name="excludeScope" value="anoninner"/>
+            <!-- Ignore errors related to unknown tags -->
+            <property name="allowUnknownTags" value="true"/>
+            <!-- Allow params tags to be optional -->
+            <property name="allowMissingParamTags" value="false"/>
+        </module>
+        <!-- Requirements for Javadocs for methods -->
+        <module name="JavadocMethod">
+            <!-- All public methods MUST HAVE Javadocs -->
+            <property name="scope" value="public"/>
+            <!-- Allow RuntimeExceptions to be undeclared -->
+            <property name="allowUndeclaredRTE" value="true"/>
+            <!-- Allow params, throws and return tags to be optional -->
+            <property name="allowMissingParamTags" value="true"/>
+            <property name="allowMissingThrowsTags" value="true"/>
+            <property name="allowMissingReturnTag" value="true"/>
+        </module>
+
+        <!-- ##### Requirements for K&R Style braces ##### -->
+        <!-- Code blocks MUST HAVE braces, even single line statements (if, while, etc) -->
+        <module name="NeedBraces"/>
+        <!-- Left braces should be at the end of current line (default value)-->
+        <module name="LeftCurly"/>
+        <!-- Right braces should be on start of a new line (default value) -->
+        <module name="RightCurly"/>
+
```

Comment:
I think we should also add some automatic checks on `switch` statements as it is easy to forget the `default` clause or to have unintended fall through behaviour:
* `<module name="MissingSwitchDefault"/>` -> http://checkstyle.sourceforge.net/config_coding.html#MissingSwitchDefault
* `<module name="FallThrough"/>` -> http://checkstyle.sourceforge.net/config_coding.html#FallThrough

Reply:
This has now been included.